content/en/docs/architecture/timeouts.md

```diff
@@ -1,24 +1,55 @@
 ---
-title: "Job Execution Timeouts"
+title: "Handling Job Timeouts and Interruptions"
 date: 2020-12-21T23:27:39-04:00
 draft: false
-description: An overview of job execution timeouts, how to configure them and what the test workload is expected to do. 
+description: An overview of job execution timeouts, how to configure them and what the test workload is expected to do when handling them. 
 ---
 
-A typical OpenShift CI job runs [ci-operator](/docs/architecture/ci-operator/) in a pod which for each [step](/docs/architecture/step-registry/#step) of the job creates a pod, executing the commands defined by `tests.steps.test.commands` of a step in the job. A _test process_ is a process which runs such commands. There are timeouts in various phases of a job execution. In this document, we will explain those timeouts, show the logs if they occur in a job, and how the test process will be notified, and how to modify the values of those timeouts.
+# Overview
 
-## Job Timeout
+Job execution timeouts and preemptions occur regularly in the OpenShift CI infrastructure, and a well-formed job must be able to handle these events. The most common tasks that jobs should execute when reacting to interruptions are to generate output artifacts and logs to aid developers investigating the job failure in the future and to clean up any shared infrastructure or cloud resources that the job may be using.
 
-The job timeout can be considered as the overall time for `ci-operator` to run. When a job fails due to the job timeout, the job log contains:
+# Architecture
 
+## How Interruptions (Preemptions and Timeouts) Occur
+
+Fully understanding the set of timeout conditions requires an overview of how jobs are executed. When a job is triggered by Prow, our top-level job orchestrator, a `Pod` is launched on the central CI build farm clusters to execute it. In OpenShift CI, this `Pod` runs [`ci-operator`](/docs/architecture/ci-operator), our job-specific orchestrator, with a cursory set of parameters (please execute job _foo_ for the _main_ branch of a specific repository). `ci-operator` in turn creates a `Namespace` on the build farm cluster in which more `Pods` execute individual [steps](/docs/architecture/step-registry/#step). Each `Pod` executes the commands as configured in `tests.steps.test.commands`, these commands are known as the _test process_. As a job author, you control the content of these commands, and the set of steps which run in the job.
```

Comment:
`build farm cluster` could link over to [here][1] or something?  I think it would be worth having somewhere in these docs that talks about jobs being sharded over a handful of build clusters, but I don't think we want that discussion to be inline here.  Maybe a follow-up PR possibility?

[1]: https://docs.ci.openshift.org/docs/how-tos/use-registries-in-build-farm/#summary-of-available-registries

Reply:
Probably best to do a topology overview in a follow-up.